src/kem/classic_mceliece/CMakeLists.txt

```diff
@@ -19,7 +19,7 @@ if(OQS_ENABLE_KEM_classic_mceliece_348864_avx2)
     add_library(classic_mceliece_348864_avx2 OBJECT pqclean_mceliece348864_avx2/aes256ctr.c pqclean_mceliece348864_avx2/benes.c pqclean_mceliece348864_avx2/bm.c pqclean_mceliece348864_avx2/consts.S pqclean_mceliece348864_avx2/controlbits.c pqclean_mceliece348864_avx2/crypto_int16.c pqclean_mceliece348864_avx2/crypto_int32.c pqclean_mceliece348864_avx2/crypto_uint16.c pqclean_mceliece348864_avx2/crypto_uint32.c pqclean_mceliece348864_avx2/crypto_uint64.c pqclean_mceliece348864_avx2/decrypt.c pqclean_mceliece348864_avx2/encrypt.c pqclean_mceliece348864_avx2/fft.c pqclean_mceliece348864_avx2/fft_tr.c pqclean_mceliece348864_avx2/gf.c pqclean_mceliece348864_avx2/operations.c pqclean_mceliece348864_avx2/pk_gen.c pqclean_mceliece348864_avx2/sk_gen.c pqclean_mceliece348864_avx2/syndrome_asm.S pqclean_mceliece348864_avx2/transpose_64x256_sp_asm.S pqclean_mceliece348864_avx2/transpose_64x64_asm.S pqclean_mceliece348864_avx2/update_asm.S pqclean_mceliece348864_avx2/vec.c pqclean_mceliece348864_avx2/vec128_mul_asm.S pqclean_mceliece348864_avx2/vec256.c pqclean_mceliece348864_avx2/vec256_mul_asm.S pqclean_mceliece348864_avx2/vec_mul_asm.S pqclean_mceliece348864_avx2/vec_mul_sp_asm.S pqclean_mceliece348864_avx2/vec_reduce_asm.S)
     target_include_directories(classic_mceliece_348864_avx2 PRIVATE ${CMAKE_CURRENT_LIST_DIR}/pqclean_mceliece348864_avx2)
     target_include_directories(classic_mceliece_348864_avx2 PRIVATE ${PROJECT_SOURCE_DIR}/src/common/pqclean_shims)
-    target_compile_options(classic_mceliece_348864_avx2 PRIVATE  -mavx2  -mpopcnt )
+    target_compile_options(classic_mceliece_348864_avx2 PRIVATE  -mavx2  -mpopcnt -O1)
```

Comment:
This reduces compiler optimization under all circumstances -- not just for memory leak testing. Is this intentional/desirable?

Reply:
This is intended. Unless we reduce the compiler optimisation for all circumstances the leak would persist for `clang` builds outside of testing.